feature/amazon/src/main/java/com/revenuecat/purchases/amazon/ProxyAmazonBillingActivityBroadcastReceiver.kt

```diff
@@ -0,0 +1,38 @@
+package com.revenuecat.purchases.amazon
+
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import androidx.annotation.VisibleForTesting
+import java.lang.ref.WeakReference
+
+internal class ProxyAmazonBillingActivityBroadcastReceiver(activity: Activity) : BroadcastReceiver() {
+
+    private val activity: WeakReference<Activity>
+
+    init {
+        this.activity = WeakReference(activity)
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    var onReceiveCalled = false
+
+    override fun onReceive(context: Context, intent: Intent) {
+        onReceiveCalled = true
+        activity.get()?.finish()
+    }
+
+    companion object {
+        const val PURCHASE_FINISHED_ACTION = "com.revenuecat.purchases.purchase_finished"
+
+        fun newPurchaseFinishedIntentFilter(): IntentFilter = IntentFilter(PURCHASE_FINISHED_ACTION)
+
+        fun newPurchaseFinishedIntent(applicationContext: Context): Intent {
+            return Intent(PURCHASE_FINISHED_ACTION).also { intent ->
+                intent.setPackage(applicationContext.packageName)
```

Comment:
Since this requires only the package name, I would probably pass that as parameter of the `newPurchaseFinishedIntent` instead of the entire context.